adr/adr-responses-api.md

```diff
@@ -0,0 +1,45 @@
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+* **Strategic Alignment & Future-Proofing:** By building on the OpenAI-compatible Responses API, we align with the industry standard and the strategic direction of Llama Stack. This de-risks our platform from the likely deprecation of the older Agents API.
+* **Ecosystem Compatibility:** Our agentic capability will be natively compatible with the vast ecosystem of tools, libraries, and monitoring solutions that are built to interface with the OpenAI API format. This accelerates both internal development and customer adoption.
+* **Enhanced Flexibility and Control:** A client-side agent construct gives the application developer full control over the agent's execution loop and tool selection. This is critical for complex, mission-critical applications where logic cannot be a black box on a server.
+* **Clear Separation of Concerns:** The Llama Stack server is responsible for what it does best: executing LLM inference with tool-calling capabilities. The client is responsible for the application-level concern of orchestrating the agentic logic. This is a clean and scalable architectural pattern.
+
+### Negative
+
+* **Implementation/Support Overhead:** We assume the responsibility for implementing/supporting whatever client-side Agent construct emerges from [ADR: Adopt a Minimal SDK and Ecosystem Strategy for Agents](https://github.com/opendatahub-io/agents/pull/2).
+* **Increased Client-side Complexity:** The logic for managing the agent's definition and execution loop now resides on the client. While our chosen Agent construct will abstract this away, the fundamental complexity is shifted from the Llama Stack server to the client. We must ensure our chosen solution is robust, performant, and easy to use. Clients must handle security, availability, and failure recovery, which becomes harder when the client owns the agent definition.  Some clients will be deployed as servers that require auto-scaling and lifecycle management.  Managing the agent object client-side can encourage this “client-as-server” pattern so multiple end-users can share the same agent, further complicating the overall architecture.
+* **Difficulty in Sharing and Collaboration:** A server-side agent can be exposed as a persistent, shareable endpoint. A client-side agent, however, exists only within the context of the application that instantiated it. To "share" an agent, one must serialize its configuration, transfer it, and have the recipient re-instantiate a new agent from that data. This "snapshot-and-restore" process creates friction for use cases requiring pre-configured, easily shareable bots or live collaborative agents. (Note: The containerized API proposal could mitigate this, which will be analyzed in the next ADR).
```

Comment:
Note that Llama Stack Agents as they are defined today can't actually be serialized, rehydrated, and used on a different server.  We would need to implement that.  There is a request from the Lightspeed teams to be able to port an agent from one Llama Stack to another in this way.  What they really want to port over is the context, configuration, and conversation memory. We can just as well implement that support in the Responses API.